File: src/main/java/io/playground/orderservice/domain/order/OrderAmount.java
package io.playground.orderservice.domain.order;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.math.BigDecimal;

@Getter
@AllArgsConstructor
public class OrderAmount {
    private Long id;

    private Long orderId;

    private BigDecimal totalAmount;

    private BigDecimal discountAmount;

    private BigDecimal deliveryFee;

    private BigDecimal finalAmount;

    public static OrderAmount of(Long id,
                                 Long orderId,
                                 BigDecimal totalAmount,
                                 BigDecimal discountAmount,
                                 BigDecimal deliveryFee,
                                 BigDecimal finalAmount) {
        return new OrderAmount(id, orderId, totalAmount, discountAmount, deliveryFee, finalAmount);
    }
}

File: src/main/kotlin/io/playground/orderservice/domain/order/OrderAmount.kt
package io.playground.orderservice.domain.order

import java.math.BigDecimal

class OrderAmount(
    val id: Long?,
    val orderId: Long?,
    val totalAmount: BigDecimal,
    val discountAmount: BigDecimal,
    val deliveryFee: BigDecimal,
    val finalAmount: BigDecimal,
) {
    companion object {
        @JvmStatic
        fun of(
            id: Long?,
            orderId: Long?,
            totalAmount: BigDecimal,
            discountAmount: BigDecimal,
            deliveryFee: BigDecimal,
            finalAmount: BigDecimal,
        ): OrderAmount = OrderAmount(id, orderId, totalAmount, discountAmount, deliveryFee, finalAmount)
    }
}
